### PR TITLE
[WIP] Generic UI Tests, Advanced Search (Bug 1380430)

### PR DESCRIPTION
**Pull request by mshriver**
# Purpose or Intent

Update web_ui.search.py to better handle clearing the filter expressions, to wait 10s for the search box to open with a fail_func, and to add some debug logging. (this can and likely should be pulled to another PR, but was useful for this new test)

**Now, the meat of the PR** - a new cfme.tests.generic_ui folder, with a new test_advanced_search module.  After discussion with @amavinag I prototyped this PR.

This test did not fit well into the other modules, and many of the advanced search tests are repeated in-large for infra.hosts, infra.vms, infra.providers.  We could consolidate and parameterize many of these tests so that advanced search is tested on all relevant pages with minimal code duplication.

I'm more than open to suggestion on a better place for tests like this to live, and if the above effort (to consolidate search tests) is worthy.

This is to automate bug 1280430, which causes all the tests to fail currently (buttons aren't found because alt-text isn't set while they're disabled).

@psav, @mmojzis, @jkrocil, @RonnyPfannschmidt, @otsuman, @dajohnso may want to weigh in.  If this is the direction we want to go, I can clean this up and expand things a bit.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 81f8729f12b1d615f6e92cffa35550113b5bdd6f:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/tests/generic_ui/test_advanced_search.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+# @pytest.mark.meta(blockers=[1380430])
+@pytest.mark.parametrize('button', search_buttons)
+def test_advanced_search_button_alt(provider, button):
+    # testing state of the buttons in the default advanced search box
+    # need to clear any existing filters, including the filter expressions, to get to default state
+    search.ensure_no_filter_applied(clear_expression=True)
+    search.ensure_advanced_search_open()
+    assert sel.is_displayed(getattr(search.search_box, button).locate)
```
Comment (RonnyPfannschmidt) [inline cfme/tests/generic_ui/test_advanced_search.py]: can you try to use the widgetastic browser here, we really want to avoid the pytest_selenium stuff in new code

at first glance we could make the search button widgets provide it

Comment (mshriver) [inline cfme/tests/generic_ui/test_advanced_search.py]: Agreed, that would absolutely be the best way to do this. Not sure it would fit into the scope here, but it could be done as one effort.

Comment (RonnyPfannschmidt) [inline cfme/tests/generic_ui/test_advanced_search.py]: @mshriver my personal preference on this is to take a look how much of a toll it would be, 

and if its more than a simple change in the invocation lines, keeping it as is now

Comment (mshriver) [inline cfme/tests/generic_ui/test_advanced_search.py]: Well, search should be refactored for widgetastic regardless. Maybe I can convince Dave for this to be the first module I convert.

**Comment by dajoRH** [comment]
Lint report for commit 02fb7b8f6e1c420ee0a21bdff892eb5d22904f28:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/generic_ui/test_advanced_search.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    navigate_to(provider, 'All')
+    assert search.is_advanced_search_possible()
+
+
+# @pytest.mark.meta(blockers=[1380430])
+@pytest.mark.parametrize('button', search_buttons)
+def test_advanced_search_button_alt(provider, button):
+    # testing state of the buttons in the default advanced search box
+    # need to clear any existing filters, including the filter expressions, to get to default state
+    search.ensure_no_filter_applied(clear_expression=True)
```
Comment (mshriver) [inline cfme/tests/generic_ui/test_advanced_search.py]: Pulled this new param from a different PR (ignore for now)
